### PR TITLE
Bump golang and alpine for security fixes

### PR DESCRIPTION
**Pull request by stafot**
Issue: MM-40235

#### Summary
Bump golang and alpine for security fixes

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40235

```release-note
Bump golang to 1.17.3 and alpine image to 3.14
```

### CODE REVIEW

**Comment by stafot** [comment]
@cpanato Could please take a look when you have some time why this pipeline fails?

**Comment by cpanato** [comment]
@stafot looks all green

File: go.mod [hunk 1]
```diff
@@ -1,6 +1,6 @@
 module github.com/mattermost/bifrost
 
-go 1.15
+go 1.16
```
Comment (cpanato) [inline go.mod:3]: why not 1.17? Are there re any clients that consume this that they are using go 1.16?

Comment (stafot) [reply go.mod:3]: It returned error that is not supported. I am not sure if I did something wrong though.

**Comment by stafot** [comment]
@cpanato Yes I managed to get to the bottom of the failures and fix them. I will check your comment about go version probably was failing on my local because I had 1.16.x.

**Pull request merged**